# Tired of Cliché Datasets? Here Are 20 Awesome Alternatives For Any Task
## TODO
![](images/unsplash.jpg)
<figcaption style="text-align: center;">
    <strong>
        Photo by 
        <a href='https://unsplash.com/@dogukan?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText'> Doğukan Şahin</a>
        on 
        <a href='https://unsplash.com/s/photos/tired?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText'>Pexels.</a> All images are by the author unless specified otherwise.
    </strong>
</figcaption>

### Setup

In [11]:
import warnings

import gdown
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import seaborn as sns
from matplotlib import rcParams

rcParams["xtick.labelsize"] = 15
rcParams["ytick.labelsize"] = 15

warnings.filterwarnings("ignore")

"I'm going to puke over my RGB backlit-keyboard *so hard* if I see one more person using Titanic, Iris, Wine or Boston datasets!"

This is the feeling you might gradually develop after being a data science learner for a while. You just can't help it - everyone wants the easy thing. Beginners use these datasets because they are stupidly straightforwad; most course creaters and bloggers use them because they are just one single Google search away.

# Regression datasets

## 1. Diamond prices and carat regression

My favorite from this list is the diamonds dataset. It is ideal in length (+50k samples) and have multiple targets you can predict as a regression or a multi-class classification task:

In [12]:
diamonds = sns.load_dataset("diamonds").sample(10).reset_index(drop=True)
diamonds

carat        cut color clarity  depth  table  price     x     y     z
0   2.15       Good     G     SI2   63.8   58.0  13317  8.23  8.08  5.20
1   1.01      Ideal     G     SI2   61.7   56.0   4260  6.47  6.43  3.98
2   0.51    Premium     E     VS2   60.3   59.0   1781  5.22  5.20  3.14
3   1.05      Ideal     F    VVS1   61.6   55.0  10872  6.57  6.53  4.04
4   0.30    Premium     G     VS1   62.5   59.0    787  4.32  4.25  2.68
5   1.06      Ideal     D     SI1   62.6   56.0   6426  6.49  6.52  4.07
6   0.25      Ideal     G     VS1   62.7   56.0    454  4.01  4.04  2.52
7   1.01      Ideal     J     SI2   61.6   55.0   3732  6.45  6.50  3.99
8   0.90  Very Good     F     SI1   62.5   59.0   3954  6.06  6.13  3.81
9   0.36      Ideal     G     VS1   60.6   57.0    761  4.58  4.62  2.79

In [13]:
sns.load_dataset("diamonds").shape

(53940, 10)

**🎯 Targets: 'carat', 'price' (regression); 'cut', 'color', 'clarity' (multi-class)**

**🔗 Link: [Kaggle](https://www.kaggle.com/shivam2503/diamondshttps://www.kaggle.com/shivam2503/diamonds)**

**📦Dimensions: (53940, 10)**

**⚙Missing values: No**

**[📚Starter notebook](https://www.kaggle.com/fuzzywizard/diamonds-in-depth-analysishttps://www.kaggle.com/fuzzywizard/diamonds-in-depth-analysis)**

## 2. Age of Abalone shells

This is a rather unique dataset from the field of zoology. The task is to predict the age of Abalone shells (a type of mollusc) using several physical measurements. Traditionally, their age are found by cutting through their cone, staining them and counting the number of rings inside the shell under a microscope. 

For zooligists, this might be fun but for data scientists, not so much:

In [14]:
abalone = pd.read_csv("data/reg6_abalone.csv")
abalone.head(10)

Sex  Length  Diameter  Height  Whole weight  Shucked weight  Viscera weight  \
0   M   0.455     0.365   0.095        0.5140          0.2245          0.1010   
1   M   0.350     0.265   0.090        0.2255          0.0995          0.0485   
2   F   0.530     0.420   0.135        0.6770          0.2565          0.1415   
3   M   0.440     0.365   0.125        0.5160          0.2155          0.1140   
4   I   0.330     0.255   0.080        0.2050          0.0895          0.0395   
5   I   0.425     0.300   0.095        0.3515          0.1410          0.0775   
6   F   0.530     0.415   0.150        0.7775          0.2370          0.1415   
7   F   0.545     0.425   0.125        0.7680          0.2940          0.1495   
8   M   0.475     0.370   0.125        0.5095          0.2165          0.1125   
9   F   0.550     0.440   0.150        0.8945          0.3145          0.1510   

   Shell weight  Rings  
0         0.150     15  
1         0.070      7  
2         0.210      9  
3         0.155     10  
4         0.055      7  
5         0.120      8  
6         0.330     20  
7         0.260     16  
8         0.165      9  
9         0.320     19

**🎯 Target: 'Rings'**

**🔗 Link: [Kaggle](https://www.kaggle.com/rodolfomendes/abalone-dataset)**

**📦Dimensions: (4177, 9)**

**⚙Missing values: No**

**[📚Starter notebook](https://www.kaggle.com/ragnisah/eda-abalone-age-prediction)**

## 3. King county house sales

This is the dataset for those who are still interested in real estate and house prices regression:

In [15]:
king = pd.read_csv("data/reg1_house_sales_king_county.csv")
king.head(10)

price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  waterfront  \
0   221900.0       3.0       1.00         1180      5650     1.0           0   
1   538000.0       3.0       2.25         2570      7242     2.0           0   
2   180000.0       2.0       1.00          770     10000     1.0           0   
3   604000.0       4.0       3.00         1960      5000     1.0           0   
4   510000.0       3.0       2.00         1680      8080     1.0           0   
5  1225000.0       4.0       4.50         5420    101930     1.0           0   
6   257500.0       3.0       2.25         1715      6819     2.0           0   
7   291850.0       3.0       1.50         1060      9711     1.0           0   
8   229500.0       3.0       1.00         1780      7470     1.0           0   
9   323000.0       3.0       2.50         1890      6560     2.0           0   

   view  condition  grade  sqft_above  yr_built  yr_renovated      lat  \
0     0          3      7        1180      1955           NaN  47.5112   
1     0          3      7        2170      1951        1991.0  47.7210   
2     0          3      6         770      1933           NaN  47.7379   
3     0          5      7        1050      1965           NaN  47.5208   
4     0          3      8        1680      1987           NaN  47.6168   
5     0          3     11        3890      2001           NaN  47.6561   
6     0          3      7        1715      1995           NaN  47.3097   
7     0          3      7        1060      1963           NaN  47.4095   
8     0          3      7        1050      1960           NaN  47.5123   
9     0          3      7        1890      2003           NaN  47.3684   

      long  sqft_living15  sqft_lot15  
0 -122.257           1340        5650  
1 -122.319           1690        7639  
2 -122.233           2720        8062  
3 -122.393           1360        5000  
4 -122.045           1800        7503  
5 -122.005           4760      101930  
6 -122.327           2238        6819  
7 -122.315           1650        9711  
8 -122.337           1780        8113  
9 -122.031           2390        7570

In [16]:
king.shape

(21613, 17)

**🎯 Target: 'price'**

**🔗 Link: [Kaggle](https://www.kaggle.com/harlfoxem/housesalesprediction)**

**📦Dimensions: (21613, 17)**

**⚙Missing values: Yes**

**[📚Starter notebook](https://www.kaggle.com/burhanykiyakoglu/predicting-house-prices)**

## 4. Cancer death rate

This dataset challenges you to find cancer mortality rate per capita (100,000) using a number of demographic variables:

In [17]:
cancer = pd.read_csv("data/reg3_cancer_death_rate.csv")
cancer.head(10)

avgAnnCount  avgDeathsPerYear  TARGET_deathRate  incidenceRate  medIncome  \
0       1397.0               469             164.9          489.8      61898   
1        173.0                70             161.3          411.6      48127   
2        102.0                50             174.7          349.7      49348   
3        427.0               202             194.8          430.4      44243   
4         57.0                26             144.4          350.1      49955   
5        428.0               152             176.0          505.4      52313   
6        250.0                97             175.9          461.8      37782   
7        146.0                71             183.6          404.0      40189   
8         88.0                36             190.5          459.4      42579   
9       4025.0              1380             177.8          510.9      60397   

   popEst2015  povertyPercent  studyPerCap           binnedInc  MedianAge  \
0      260131            11.2   499.748204   (61494.5, 125635]       39.3   
1       43269            18.6    23.111234  (48021.6, 51046.4]       33.0   
2       21026            14.6    47.560164  (48021.6, 51046.4]       45.0   
3       75882            17.1   342.637253    (42724.4, 45201]       42.8   
4       10321            12.5     0.000000  (48021.6, 51046.4]       48.3   
5       61023            15.6   180.259902  (51046.4, 54545.6]       45.4   
6       41516            23.2     0.000000  (37413.8, 40362.7]       42.6   
7       20848            17.8     0.000000  (37413.8, 40362.7]       51.7   
8       13088            22.3     0.000000  (40362.7, 42724.4]       49.3   
9      843954            13.1   427.748432  (54545.6, 61494.5]       35.8   

   ...  PctPrivateCoverageAlone  PctEmpPrivCoverage  PctPublicCoverage  \
0  ...                      NaN                41.6               32.9   
1  ...                     53.8                43.6               31.1   
2  ...                     43.5                34.9               42.1   
3  ...                     40.3                35.0               45.3   
4  ...                     43.9                35.1               44.0   
5  ...                     38.8                32.6               43.2   
6  ...                     35.0                28.3               46.4   
7  ...                     33.1                25.9               50.9   
8  ...                     37.8                29.9               48.1   
9  ...                      NaN                44.4               31.4   

   PctPublicCoverageAlone   PctWhite  PctBlack  PctAsian  PctOtherRace  \
0                    14.0  81.780529  2.594728  4.821857      1.843479   
1                    15.3  89.228509  0.969102  2.246233      3.741352   
2                    21.1  90.922190  0.739673  0.465898      2.747358   
3                    25.0  91.744686  0.782626  1.161359      1.362643   
4                    22.7  94.104024  0.270192  0.665830      0.492135   
5                    20.2  84.882631  1.653205  1.538057      3.314635   
6                    28.7  75.106455  0.616955  0.866157      8.356721   
7                    24.1  89.406636  0.305159  1.889077      2.286268   
8                    26.6  91.787477  0.185071  0.208205      0.616903   
9                    16.5  74.729668  6.710854  6.041472      2.699184   

   PctMarriedHouseholds  BirthRate  
0             52.856076   6.118831  
1             45.372500   4.333096  
2             54.444868   3.729488  
3             51.021514   4.603841  
4             54.027460   6.796657  
5             51.220360   4.964476  
6             51.013900   4.204317  
7             48.967033   5.889179  
8             53.446998   5.587583  
9             50.063573   5.533430  

[10 rows x 33 columns]

**🎯 Target: 'TARGET_deathRate'**

**🔗 Link: [Data.world](https://data.world/nrippner/ols-regression-challenge)**

**📦Dimensions: (3047, 33)**

**⚙Missing values: Yes**

## 5. Life expectancy

How long a person will live? This is one of the hardest questions unanswered in science because it is hard to predict. Several studies have been undertaken to understand human life and this dataset provided by WHO (World Health Orginazation) is one of them:

In [18]:
who = pd.read_csv("data/reg4_life_expectancy.csv")
who.head(10)

Year      Status  Life expectancy   Adult Mortality  infant deaths  \
0  2015  Developing              65.0            263.0             62   
1  2014  Developing              59.9            271.0             64   
2  2013  Developing              59.9            268.0             66   
3  2012  Developing              59.5            272.0             69   
4  2011  Developing              59.2            275.0             71   
5  2010  Developing              58.8            279.0             74   
6  2009  Developing              58.6            281.0             77   
7  2008  Developing              58.1            287.0             80   
8  2007  Developing              57.5            295.0             82   
9  2006  Developing              57.3            295.0             84   

   Alcohol  percentage expenditure  Hepatitis B  Measles    BMI   ...  Polio  \
0     0.01               71.279624         65.0      1154   19.1  ...    6.0   
1     0.01               73.523582         62.0       492   18.6  ...   58.0   
2     0.01               73.219243         64.0       430   18.1  ...   62.0   
3     0.01               78.184215         67.0      2787   17.6  ...   67.0   
4     0.01                7.097109         68.0      3013   17.2  ...   68.0   
5     0.01               79.679367         66.0      1989   16.7  ...   66.0   
6     0.01               56.762217         63.0      2861   16.2  ...   63.0   
7     0.03               25.873925         64.0      1599   15.7  ...   64.0   
8     0.02               10.910156         63.0      1141   15.2  ...   63.0   
9     0.03               17.171518         64.0      1990   14.7  ...   58.0   

   Total expenditure  Diphtheria    HIV/AIDS         GDP  Population  \
0               8.16         65.0        0.1  584.259210  33736494.0   
1               8.18         62.0        0.1  612.696514    327582.0   
2               8.13         64.0        0.1  631.744976  31731688.0   
3               8.52         67.0        0.1  669.959000   3696958.0   
4               7.87         68.0        0.1   63.537231   2978599.0   
5               9.20         66.0        0.1  553.328940   2883167.0   
6               9.42         63.0        0.1  445.893298    284331.0   
7               8.33         64.0        0.1  373.361116   2729431.0   
8               6.73         63.0        0.1  369.835796  26616792.0   
9               7.43         58.0        0.1  272.563770   2589345.0   

    thinness  1-19 years   thinness 5-9 years  \
0                   17.2                 17.3   
1                   17.5                 17.5   
2                   17.7                 17.7   
3                   17.9                 18.0   
4                   18.2                 18.2   
5                   18.4                 18.4   
6                   18.6                 18.7   
7                   18.8                 18.9   
8                   19.0                 19.1   
9                   19.2                 19.3   

   Income composition of resources  Schooling  
0                            0.479       10.1  
1                            0.476       10.0  
2                            0.470        9.9  
3                            0.463        9.8  
4                            0.454        9.5  
5                            0.448        9.2  
6                            0.434        8.9  
7                            0.433        8.7  
8                            0.415        8.4  
9                            0.405        8.1  

[10 rows x 21 columns]

**🎯 Target: 'Life expectancy'**

**🔗 Link: [Kaggle](https://www.kaggle.com/kumarajarshi/life-expectancy-who/)**

**📦Dimensions: (2938, 21)**

**⚙Missing values: Yes**

**[📚Starter notebook](https://www.kaggle.com/mathchi/life-expectancy-who-with-several-ml-techniques)**

## 6. Car prices

The title says it all - predict car prices using variables like mileage, fuel type, transmission and several domain-specific features. This is also an excellent dataset for pumping out your feature engineering muscles:

In [19]:
cars = pd.read_csv("data/reg5_vehicle_prices.csv")
cars.head(10)

year  selling_price  km_driven    fuel seller_type transmission  \
0  2014         450000     145500  Diesel  Individual       Manual   
1  2014         370000     120000  Diesel  Individual       Manual   
2  2006         158000     140000  Petrol  Individual       Manual   
3  2010         225000     127000  Diesel  Individual       Manual   
4  2007         130000     120000  Petrol  Individual       Manual   
5  2017         440000      45000  Petrol  Individual       Manual   
6  2007          96000     175000     LPG  Individual       Manual   
7  2001          45000       5000  Petrol  Individual       Manual   
8  2011         350000      90000  Diesel  Individual       Manual   
9  2013         200000     169000  Diesel  Individual       Manual   

          owner  mileage   engine  max_power                    torque  seats  
0   First Owner     23.0  1248 CC       74.0            190Nm@ 2000rpm    5.0  
1  Second Owner     21.0  1498 CC      103.0       250Nm@ 1500-2500rpm    5.0  
2   Third Owner     17.0  1497 CC       78.0     12.7@ 2,700(kgm@ rpm)    5.0  
3   First Owner     23.0  1396 CC       90.0  22.4 kgm at 1750-2750rpm    5.0  
4   First Owner     16.0  1298 CC       88.0     11.5@ 4,500(kgm@ rpm)    5.0  
5   First Owner     20.0  1197 CC       81.0         113.75nm@ 4000rpm    5.0  
6   First Owner     17.0  1061 CC       57.0      7.8@ 4,500(kgm@ rpm)    5.0  
7  Second Owner     16.0   796 CC       37.0             59Nm@ 2500rpm    4.0  
8   First Owner     23.0  1364 CC       67.0       170Nm@ 1800-2400rpm    5.0  
9   First Owner     20.0  1399 CC       68.0            160Nm@ 2000rpm    5.0

**🎯 Target: 'selling_price'**

**🔗 Link: [Kaggle](https://www.kaggle.com/nehalbirla/vehicle-dataset-from-cardekho?ref=hackernoon.com&select=Car+details+v3.csv)**

**📦Dimensions: (8128, 12)**

**⚙Missing values: Yes**

**[📚Starter notebook](https://www.kaggle.com/mohaiminul101/car-price-prediction)**

# Binary classification

## 7. NBA rookie stats

The first binary classification dataset in the list requires you to predict if a rookie basketball player will last more than 5 years in the league:

In [20]:
nba = pd.read_csv("data/bin1_nba.csv")
nba.head()

GP   MIN  PTS  FGM  FGA   FG%  3P Made  3PA   3P%  FTM  FTA   FT%  OREB  \
0  36  27.4  7.4  2.6  7.6  34.7      0.5  2.1  25.0  1.6  2.3  69.9   0.7   
1  35  26.9  7.2  2.0  6.7  29.6      0.7  2.8  23.5  2.6  3.4  76.5   0.5   
2  74  15.3  5.2  2.0  4.7  42.2      0.4  1.7  24.4  0.9  1.3  67.0   0.5   
3  58  11.6  5.7  2.3  5.5  42.6      0.1  0.5  22.6  0.9  1.3  68.9   1.0   
4  48  11.5  4.5  1.6  3.0  52.4      0.0  0.1   0.0  1.3  1.9  67.4   1.0   

   DREB  REB  AST  STL  BLK  TOV  TARGET_5Yrs  
0   3.4  4.1  1.9  0.4  0.4  1.3          0.0  
1   2.0  2.4  3.7  1.1  0.5  1.6          0.0  
2   1.7  2.2  1.0  0.5  0.3  1.0          0.0  
3   0.9  1.9  0.8  0.6  0.1  1.0          1.0  
4   1.5  2.5  0.3  0.3  0.4  0.8          1.0

**🎯 Target: 'TARGET_5Yrs'**

**🔗 Link: [Data.world](https://data.world/exercises/logistic-regression-exercise-1)**

**📦Dimensions: (8128, 12)**

**⚙Missing values: Yes**

**[📚Starter notebook](https://www.kaggle.com/mohaiminul101/car-price-prediction)**

## 8. Stroke prediction

Another medical dataset in the list asks you to predict whether a patient will have a stroke or not based on their history. Very interesting dataset:

In [21]:
stroke = pd.read_csv("data/bin2_stroke.csv")
stroke.head()

gender   age  hypertension  heart_disease ever_married      work_type  \
0    Male  67.0             0              1          Yes        Private   
1  Female  61.0             0              0          Yes  Self-employed   
2    Male  80.0             0              1          Yes        Private   
3  Female  49.0             0              0          Yes        Private   
4  Female  79.0             1              0          Yes  Self-employed   

  Residence_type  avg_glucose_level   bmi   smoking_status  stroke  
0          Urban             228.69  36.6  formerly smoked       1  
1          Rural             202.21   NaN     never smoked       1  
2          Rural             105.92  32.5     never smoked       1  
3          Urban             171.23  34.4           smokes       1  
4          Rural             174.12  24.0     never smoked       1

**🎯 Target: 'stroke'**

**🔗 Link: [Kaggle](https://www.kaggle.com/fedesoriano/stroke-prediction-dataset)**

**📦Dimensions: (5110, 11)**

**⚙Missing values: Yes**

**[📚Starter notebook](https://www.kaggle.com/joshuaswords/predicting-a-stroke-shap-lime-explainer-eli5)**

## 9. Water potability

Safe drinking water is the most basic human right and a major influencer on health. Using this dataset, you should classify water boides into potable (drinkable) and not potable using a number of chemical properties:

In [2]:
water = pd.read_csv("data/bin3_water.csv")
water.head(10)

<IPython.core.display.Javascript object>

ph    Hardness        Solids  Chloramines     Sulfate  Conductivity  \
0        NaN  204.890455  20791.318981     7.300212  368.516441    564.308654   
1   3.716080  129.422921  18630.057858     6.635246         NaN    592.885359   
2   8.099124  224.236259  19909.541732     9.275884         NaN    418.606213   
3   8.316766  214.373394  22018.417441     8.059332  356.886136    363.266516   
4   9.092223  181.101509  17978.986339     6.546600  310.135738    398.410813   
5   5.584087  188.313324  28748.687739     7.544869  326.678363    280.467916   
6  10.223862  248.071735  28749.716544     7.513408  393.663396    283.651634   
7   8.635849  203.361523  13672.091764     4.563009  303.309771    474.607645   
8        NaN  118.988579  14285.583854     7.804174  268.646941    389.375566   
9  11.180284  227.231469  25484.508491     9.077200  404.041635    563.885481   

   Organic_carbon  Trihalomethanes  Turbidity  Potability  
0       10.379783        86.990970   2.963135           0  
1       15.180013        56.329076   4.500656           0  
2       16.868637        66.420093   3.055934           0  
3       18.436524       100.341674   4.628771           0  
4       11.558279        31.997993   4.075075           0  
5        8.399735        54.917862   2.559708           0  
6       13.789695        84.603556   2.672989           0  
7       12.363817        62.798309   4.401425           0  
8       12.706049        53.928846   3.595017           0  
9       17.927806        71.976601   4.370562           0

**🎯 Target: 'Potability'**

**🔗 Link: [Kaggle](https://www.kaggle.com/adityakadiwal/water-potability)**

**📦Dimensions: (3276, 10)**

**⚙Missing values: Yes**

**[📚Starter notebook](https://www.kaggle.com/jaykumar1607/water-quality-analysis-plotly-and-modelling)**

## 10. Smart grid stability

This is an augmented version of "Electrical Grid Stability Simulated Dataset" created by Vadim Arzamasov and donated to UCI and made available on Kaggle as well. You will be predicting the stability of 4-node smart grid systems (whatever they mean):

In [5]:
grid = pd.read_csv("data/bin4_grid_stability.csv")
grid.head(10)

<IPython.core.display.Javascript object>

tau1      tau2      tau3      tau4        p1        p2        p3  \
0  2.959060  3.079885  8.381025  9.780754  3.763085 -0.782604 -1.257395   
1  9.304097  4.902524  3.047541  1.369357  5.067812 -1.940058 -1.872742   
2  8.971707  8.848428  3.046479  1.214518  3.405158 -1.207456 -1.277210   
3  0.716415  7.669600  4.486641  2.340563  3.963791 -1.027473 -1.938944   
4  3.134112  7.608772  4.943759  9.857573  3.525811 -1.125531 -1.845975   
5  6.999209  9.109247  3.784066  4.267788  4.429669 -1.857139 -0.670397   
6  6.710166  3.765204  6.929314  8.818562  2.397419 -0.614590 -1.208826   
7  6.953512  1.379125  5.719400  7.870307  3.224495 -0.748998 -1.186517   
8  4.689852  4.007747  1.478573  3.733787  4.041300 -1.410344 -1.238204   
9  9.841496  1.413822  9.769856  7.641616  4.727595 -1.991363 -0.857637   

         p4        g1        g2        g3        g4  stabf  
0 -1.723086  0.650456  0.859578  0.887445  0.958034      0  
1 -1.255012  0.413441  0.862414  0.562139  0.781760      1  
2 -0.920492  0.163041  0.766689  0.839444  0.109853      0  
3 -0.997374  0.446209  0.976744  0.929381  0.362718      0  
4 -0.554305  0.797110  0.455450  0.656947  0.820923      0  
5 -1.902133  0.261793  0.077930  0.542884  0.469931      1  
6 -0.574004  0.177890  0.397977  0.402046  0.376630      0  
7 -1.288980  0.371385  0.633204  0.732741  0.380544      0  
8 -1.392751  0.269708  0.250364  0.164941  0.482439      1  
9 -1.878594  0.376356  0.544415  0.792039  0.116263      0

**🎯 Target: 'stabf'**

**🔗 Link: [Kaggle](https://www.kaggle.com/pcbreviglieri/smart-grid-stability)**

**📦Dimensions: (60000, 13)**

**⚙Missing values: No**

**[📚Starter notebook](https://www.kaggle.com/pcbreviglieri/predicting-smart-grid-stability-with-deep-learning)**

## 11. IBM HR analytics & employee attrition

This fictional dataset created by IBM datasets tasks you to uncover which factors lead to employee attrition (whether they will leave their role):

In [8]:
hr = pd.read_csv("data/attrition.csv")
hr.head(10)

<IPython.core.display.Javascript object>

Age Attrition     BusinessTravel  DailyRate              Department  \
0   41       Yes      Travel_Rarely       1102                   Sales   
1   49        No  Travel_Frequently        279  Research & Development   
2   37       Yes      Travel_Rarely       1373  Research & Development   
3   33        No  Travel_Frequently       1392  Research & Development   
4   27        No      Travel_Rarely        591  Research & Development   
5   32        No  Travel_Frequently       1005  Research & Development   
6   59        No      Travel_Rarely       1324  Research & Development   
7   30        No      Travel_Rarely       1358  Research & Development   
8   38        No  Travel_Frequently        216  Research & Development   
9   36        No      Travel_Rarely       1299  Research & Development   

   DistanceFromHome  Education EducationField  EmployeeCount  EmployeeNumber  \
0                 1          2  Life Sciences              1               1   
1                 8          1  Life Sciences              1               2   
2                 2          2          Other              1               4   
3                 3          4  Life Sciences              1               5   
4                 2          1        Medical              1               7   
5                 2          2  Life Sciences              1               8   
6                 3          3        Medical              1              10   
7                24          1  Life Sciences              1              11   
8                23          3  Life Sciences              1              12   
9                27          3        Medical              1              13   

   ...  RelationshipSatisfaction StandardHours  StockOptionLevel  \
0  ...                         1            80                 0   
1  ...                         4            80                 1   
2  ...                         2            80                 0   
3  ...                         3            80                 0   
4  ...                         4            80                 1   
5  ...                         3            80                 0   
6  ...                         1            80                 3   
7  ...                         2            80                 1   
8  ...                         2            80                 0   
9  ...                         2            80                 2   

   TotalWorkingYears  TrainingTimesLastYear WorkLifeBalance  YearsAtCompany  \
0                  8                      0               1               6   
1                 10                      3               3              10   
2                  7                      3               3               0   
3                  8                      3               3               8   
4                  6                      3               3               2   
5                  8                      2               2               7   
6                 12                      3               2               1   
7                  1                      2               3               1   
8                 10                      2               3               9   
9                 17                      3               2               7   

  YearsInCurrentRole  YearsSinceLastPromotion  YearsWithCurrManager  
0                  4                        0                     5  
1                  7                        1                     7  
2                  0                        0                     0  
3                  7                        3                     0  
4                  2                        2                     2  
5                  7                        3                     6  
6                  0                        0                     0  
7                  0                        0                     0  
8                  7                        1             

**🎯 Target: 'Attrition'**

**🔗 Link: [Kaggle](https://www.kaggle.com/pavansubhasht/ibm-hr-analytics-attrition-dataset)**

**📦Dimensions: (1470, 35)**

**⚙Missing values: No**

**[📚Starter notebook](https://www.kaggle.com/janiobachmann/attrition-in-an-organization-why-workers-quit)**


## 12. Can I eat this mushroom?

Another one of a kind dataset is classifying mushrooms into edible and poisonous. It also presents a unique challenge - all features are categorical:

In [11]:
shroom = pd.read_csv("data/bin7_mushroom.csv")
shroom.head(10)

<IPython.core.display.Javascript object>

class cap-shape cap-surface cap-color bruises odor gill-attachment  \
0     p         x           s         n       t    p               f   
1     e         x           s         y       t    a               f   
2     e         b           s         w       t    l               f   
3     p         x           y         w       t    p               f   
4     e         x           s         g       f    n               f   
5     e         x           y         y       t    a               f   
6     e         b           s         w       t    a               f   
7     e         b           y         w       t    l               f   
8     p         x           y         w       t    p               f   
9     e         b           s         y       t    a               f   

  gill-spacing gill-size gill-color  ... stalk-surface-below-ring  \
0            c         n          k  ...                        s   
1            c         b          k  ...                        s   
2            c         b          n  ...                        s   
3            c         n          n  ...                        s   
4            w         b          k  ...                        s   
5            c         b          n  ...                        s   
6            c         b          g  ...                        s   
7            c         b          n  ...                        s   
8            c         n          p  ...                        s   
9            c         b          g  ...                        s   

  stalk-color-above-ring stalk-color-below-ring veil-type veil-color  \
0                      w                      w         p          w   
1                      w                      w         p          w   
2                      w                      w         p          w   
3                      w                      w         p          w   
4                      w                      w         p          w   
5                      w                      w         p          w   
6                      w                      w         p          w   
7                      w                      w         p          w   
8                      w                      w         p          w   
9                      w                      w         p          w   

  ring-number ring-type spore-print-color population habitat  
0           o         p                 k          s       u  
1           o         p                 n          n       g  
2           o         p                 n          n       m  
3           o         p                 k          s       u  
4           o         e                 n          a       g  
5           o         p                 k          n       g  
6           o         p                 k          n       m  
7           o         p                 n          s       m  
8           o         p                 k          v       g  
9           o         p                 k          s       m  

[10 rows x 23 columns]

**🎯 Target: 'class'**

**🔗 Link: [Kaggle](https://www.kaggle.com/uciml/mushroom-classification)**

**📦Dimensions: (8124, 23)**

**⚙Missing values: Yes**

**[📚Starter notebook](https://www.kaggle.com/alincijov/mushroom-classification-using-genetic-algorithm)**

## 13. Banknote authentication

Even though this dataset has very few features, I wanted to include it because the task is really interesting - using physical attributes of banknotes, you should classify them into forged or original:

In [19]:
note = pd.read_csv("data/banknote.csv")
note.head(10)

<IPython.core.display.Javascript object>

variance  skewness  curtosis  entropy  class
0   3.62160    8.6661  -2.80730 -0.44699      0
1   4.54590    8.1674  -2.45860 -1.46210      0
2   3.86600   -2.6383   1.92420  0.10645      0
3   3.45660    9.5228  -4.01120 -3.59440      0
4   0.32924   -4.4552   4.57180 -0.98880      0
5   4.36840    9.6718  -3.96060 -3.16250      0
6   3.59120    3.0129   0.72888  0.56421      0
7   2.09220   -6.8100   8.46360 -0.60216      0
8   3.20320    5.7588  -0.75345 -0.61251      0
9   1.53560    9.1772  -2.27180 -0.73535      0

**🎯 Target: 'class'**

**🔗 Link: [Kaggle](https://www.kaggle.com/ritesaluja/bank-note-authentication-uci-data)**

**📦Dimensions: (1372, 5)**

**⚙Missing values: No**

**[📚Starter notebook](https://www.kaggle.com/vivekgediya/banknote-authentication-analysis)**

## 14. Adult income dataset

Predict whether a person will end up earning more than 50k or not using factors like their age, education, backgroud, gender, marital status, etc.:

In [22]:
adult = pd.read_csv("data/bin6_adult_income.csv")
adult.head(10)

<IPython.core.display.Javascript object>

age         workclass  fnlwgt     education  educational-num  \
0   25           Private  226802          11th                7   
1   38           Private   89814       HS-grad                9   
2   28         Local-gov  336951    Assoc-acdm               12   
3   44           Private  160323  Some-college               10   
4   18               NaN  103497  Some-college               10   
5   34           Private  198693          10th                6   
6   29               NaN  227026       HS-grad                9   
7   63  Self-emp-not-inc  104626   Prof-school               15   
8   24           Private  369667  Some-college               10   
9   55           Private  104996       7th-8th                4   

       marital-status         occupation   relationship   race  gender  \
0       Never-married  Machine-op-inspct      Own-child  Black    Male   
1  Married-civ-spouse    Farming-fishing        Husband  White    Male   
2  Married-civ-spouse    Protective-serv        Husband  White    Male   
3  Married-civ-spouse  Machine-op-inspct        Husband  Black    Male   
4       Never-married                NaN      Own-child  White  Female   
5       Never-married      Other-service  Not-in-family  White    Male   
6       Never-married                NaN      Unmarried  Black    Male   
7  Married-civ-spouse     Prof-specialty        Husband  White    Male   
8       Never-married      Other-service      Unmarried  White  Female   
9  Married-civ-spouse       Craft-repair        Husband  White    Male   

   capital-gain  capital-loss  hours-per-week native-country  income  
0             0             0              40  United-States       0  
1             0             0              50  United-States       0  
2             0             0              40  United-States       1  
3          7688             0              40  United-States       1  
4             0             0              30  United-States       0  
5             0             0              30  United-States       0  
6             0             0              40  United-States       0  
7          3103             0              32  United-States       1  
8             0             0              40  United-States       0  
9             0             0              10  United-States       0

**🎯 Target: 'inco,e'**

**🔗 Link: [Kaggle](https://www.kaggle.com/wenruliu/adult-income-dataset)**

**📦Dimensions: (48842, 15)**

**⚙Missing values: Yes**

**[📚Starter notebook](https://www.kaggle.com/alokevil/simple-eda-for-beginners)**

# Mutli-class classification datasets

> Here is a [handy tutorial](https://towardsdatascience.com/comprehensive-guide-to-multiclass-classification-with-sklearn-127cc500f362?source=your_stories_page-------------------------------------) on multi-class classification and a comprehensive guide on [mutli-class classification metrics](https://towardsdatascience.com/comprehensive-guide-on-multiclass-classification-metrics-af94cfb83fbd?source=your_stories_page-------------------------------------). 

## 16. Yeast classification

This dataset will give you a small taste from the world of microbiology. You are tasked to classify a fungus called yeast into species:

In [25]:
yeast = pd.read_csv("data/m4_yeast.csv")
yeast.head(10)

<IPython.core.display.Javascript object>

mcg   gvh   alm   mit  erl  pox   vac   nuc  class_protein_localization
0  0.58  0.61  0.47  0.13  0.5  0.0  0.48  0.22                           0
1  0.43  0.67  0.48  0.27  0.5  0.0  0.53  0.22                           0
2  0.64  0.62  0.49  0.15  0.5  0.0  0.53  0.22                           0
3  0.58  0.44  0.57  0.13  0.5  0.0  0.54  0.22                           1
4  0.42  0.44  0.48  0.54  0.5  0.0  0.48  0.22                           0
5  0.51  0.40  0.56  0.17  0.5  0.5  0.49  0.22                           2
6  0.50  0.54  0.48  0.65  0.5  0.0  0.53  0.22                           0
7  0.48  0.45  0.59  0.20  0.5  0.0  0.58  0.34                           1
8  0.55  0.50  0.66  0.36  0.5  0.0  0.49  0.22                           0
9  0.40  0.39  0.60  0.15  0.5  0.0  0.58  0.30                           2

**🎯 Target: 'class_protein_localization'**

**🔗 Link: [OpenML](https://www.openml.org/d/181)**

**📦Dimensions: (1484, 9)**

**⚙Missing values: No**

## 17. Kaggle TPS May 2021

Kaggle hosts monthly competitions called "Tabular Playground Series" with a beginner-to-medium difficult tasks. The most important point is that each month a new, large-sized synthetic dataset is created using CTGAN framework. This one is from the May edition:

In [28]:
tps_may = pd.read_csv("data/m5_tps_may.csv")
tps_may.head(10)

<IPython.core.display.Javascript object>

id  feature_0  feature_1  feature_2  feature_3  feature_4  feature_5  \
0   0          0          0          1          0          1          0   
1   1          0          0          0          0          2          1   
2   2          0          0          0          0          0          0   
3   3          0          0          0          0          0          0   
4   4          0          0          0          0          0          0   
5   5          0          0          0          0          0          0   
6   6          0          0          0          1          0          0   
7   7          0          0          0          2          0          0   
8   8          0          1          0          0          0          0   
9   9          0          0          1          0          0          0   

   feature_6  feature_7  feature_8  ...  feature_41  feature_42  feature_43  \
0          0          0          0  ...           0           0          21   
1          0          0          0  ...           0           0           0   
2          0          0          0  ...           0           1           0   
3          0          3          0  ...           0           0           0   
4          0          0          0  ...           0           0           0   
5          0          0          0  ...           0           0           0   
6          0          0          0  ...           0           0           0   
7          3          0          0  ...           0           1           0   
8          0          0         15  ...           0           0           0   
9          0          0          1  ...           1           0           0   

   feature_44  feature_45  feature_46  feature_47  feature_48  feature_49  \
0           0           0           0           0           0           0   
1           0           0           0           0           0           0   
2           0           0           0          13           2           0   
3           0           0           0           0           1           0   
4           0           0           0           0           1           0   
5           0           0           0           0           0           1   
6           0           0           2           0           0           0   
7           0           0           0           2           0           0   
8           0           0           0           0           0           1   
9           0           0           0           0           1           0   

   target  
0       0  
1       1  
2       1  
3       2  
4       0  
5       0  
6       1  
7       0  
8       0  
9       2  

[10 rows x 52 columns]

**🎯 Target: 'target'**

**🔗 Link: [Kaggle](https://www.kaggle.com/nehalbirla/vehicle-dataset-from-cardekho?ref=hackernoon.com&select=Car+details+v3.csv)**

**📦Dimensions: (100000, 52)**

**⚙Missing values: No**

**[📚Starter notebook](https://www.kaggle.com/subinium/tps-may-categorical-eda)**